changelog/pending/20241003--cli--allow-memory-profile-to-be-written-occasionally-in-case-of-crash.yaml

```
@@ -0,0 +1,4 @@
+changes:
+- type: feat
+  scope: cli
+  description: allow memory profile to be written occasionally in case of crash
```

```suggestion
  description: Allow memory profile to be written in the background, so it's available in case of crash
```

I think that's what we're trying to achieve here.  Otherwise it kinda sounds like we sometimes write it when there is a crash to me.